File: clients/typescript/examples/collections-insert-many-blob.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Insert a document with a binary field
(async function () {
  const result = await collection.insertMany([
    {
      exampleBinary: { $binary: "PfvnbT7peNU/Sfvn" },
    },
  ]);
})();

File: clients/python/examples/collections-insert-many-blob.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Insert a document with binary fields
result = collection.insert_many(
    [
        {
            "exampleBinary": {"$binary": "PfvnbT7peNU/Sfvn"},
            "anotherExampleBinary": b"=\xfb\xe7m>\xe9x\xd5?I\xfb\xe7",
        }
    ]
)
